project/CliOptions.scala

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka
+
+case class CliOption[T](value: T)
+
+object CliOption {
+  def apply[T](path: String, default: T)(implicit ev: CliOptionParser[T]): CliOption[T] = ev.parse(path, default)
+
+  implicit class BooleanCliOption(cliOption: CliOption[Boolean]) {
```

Comment:
WDYM? Extending `AnyVal` (and making `val cliOption`) would mean that this erases to `CliOption` and is stored as that without a wrapper. This is not performance sensitive code, so it is not needed, it is just a habit/pattern to model extension methods using implicit value classes. The only reason for not doing this would be if this were declared within a trait (because then it does not work).